container/sagemaker/python_service.py

```diff
@@ -128,11 +133,56 @@ def on_get(self, req, res):  # pylint: disable=W0613
         res.status = falcon.HTTP_200
 
 
-# receiving requests to /invocations and /ping
-ping_resource = PingResource()
-invocation_resource = InvocationResource()
+class ModelManagerResource(object):
 
-app = falcon.API()
+    def __init__(self):
+        self.grpc_client = GRPCProxyClient(TFS_GRPC_PORT)
+
+    def on_get(self, req, res):  # pylint: disable=W0613
+        with open('/sagemaker/model-config.cfg') as f:
+            models = f.read()
+            res.body = models
+            res.status = falcon.HTTP_200
 
-app.add_route('/ping', ping_resource)
-app.add_route('/invocations', invocation_resource)
+    def on_post(self, req, res):
+        res.status = falcon.HTTP_200
+        data = json.loads(req.stream.read().decode('utf-8'))
+        model_name = data['name']
+        base_path = data['uri']
+        try:
+            # prevent concurrent load model requests
+            lock_file = open(MODEL_CONFIG_FILE_PATH, 'w')
```

Comment:
Is it possible to prevent the concurrent requests on NGINX or GUnicorn?

Reply:
Yes, I used nginx to limit connection for /models